File: app/src/main/java/xyz/lrhm/komakdast/ui/lessonScreen/components/NextLevelDialog.java
package xyz.lrhm.komakdast.ui.lessonScreen.components;

import android.app.Dialog;
import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.util.TypedValue;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.constraintlayout.widget.ConstraintLayout;

import com.bumptech.glide.Glide;

import xyz.lrhm.komakdast.R;
import xyz.lrhm.komakdast.core.data.model.Lesson;
import xyz.lrhm.komakdast.core.util.legacy.LengthManager;

public class NextLevelDialog extends Dialog implements View.OnClickListener {
    private Context context;
    private Lesson level;
    private int packageSize;
    private int prize;
    private FinishLevel finishLevel;
    private LengthManager lengthManager;
    private boolean skiped;

    public NextLevelDialog(Context context, Lesson level, int packageSize, boolean skiped, int prize, FinishLevel finishLevel) {
        super(context);
        this.context = context;
        this.level = level;
        this.packageSize = packageSize + 1;
        this.prize = prize;
        this.finishLevel = finishLevel;
        this.skiped = skiped;
        lengthManager = new LengthManager(context);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setCanceledOnTouchOutside(true);


        setContentView(R.layout.dialog_next_level);

        getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        TextView prizeTextView = (TextView) findViewById(R.id.prizeTextView);

        ImageView scarf = findViewById(R.id.scarf_image_view);

        if (!level.getResolved() && !skiped) {
            String prizeString;
            if (prize == 30)
                prizeString = "+۳۰";
            else if (prize == 10)
                prizeString = "+۱۰";
            else
                prizeString = "+۵";

            customizeTextView(prizeTextView, prizeString, lengthManager.getLevelAuthorTextSize());

            Glide.with(getContext()).load(R.drawable.scarf).into(scarf);


        } else {
            prizeTextView.setVisibility(View.GONE);

            scarf.setVisibility(View.GONE);
        }

        int width = (int) (lengthManager.getScreenWidth() * 0.8f);
        int height = lengthManager.getHeightWithFixedWidth(

                R.drawable.dialog_background, (int) (lengthManager.getScreenWidth() * 0.8f)
        );

        ConstraintLayout container = findViewById(R.id.dialog_container);


//        container.setMaxWidth(width);
        container.setMinWidth(width);
        container.setMinHeight(height);

        ImageView background = findViewById(R.id.dialog_background);
        ConstraintLayout.LayoutParams lp = (ConstraintLayout.LayoutParams) background.getLayoutParams();
        lp.width = width;
        lp.height = height;
        background.setLayoutParams(lp);
//        UiUtil.setWidth(background, width);
//        UiUtil.setHeight(background, height);


        Glide.with(getContext()).load(R.drawable.dialog_background).into(background);


        ImageView nextButton = findViewById(R.id.next_level_button);
        Glide.with(getContext()).load(R.drawable.next_button_dialog).into(nextButton);

        nextButton.setOnClickListener(this);

        ImageView homeButton = findViewById(R.id.home_button);
        Glide.with(getContext()).load(R.drawable.hoem_button).into(homeButton);

        homeButton.setOnClickListener(this);

        ImageView tick = findViewById(R.id.tick);
        Glide.with(getContext()).load(R.drawable.tick).into(tick);

    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.home_button:
                finishLevel.Home();
                break;

            case R.id.next_level_button:
                finishLevel.NextLevel();
                break;
        }

        dismiss();
    }

    private void customizeTextView(TextView textView, String label, float textSize) {
        textView.setText(label);
        textView.setTextSize(TypedValue.COMPLEX_UNIT_PX, lengthManager.getStoreItemFontSize());
        textView.setTextColor(Color.WHITE);
        textView.setShadowLayer(1, 2, 2, Color.BLACK);
//        textView.setTypeface(FontsHolder.getSansBold(textView.getContext()));
    }
}

File: app/src/main/java/xyz/lrhm/komakdast/ui/lessonScreen/components/NextLevelDialog.kt
package xyz.lrhm.komakdast.ui.lessonScreen.components

import android.app.Dialog
import android.content.Context
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.util.TypedValue
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import com.bumptech.glide.Glide
import xyz.lrhm.komakdast.R
import xyz.lrhm.komakdast.core.data.model.Lesson
import xyz.lrhm.komakdast.core.util.legacy.LengthManager

class NextLevelDialog(
    context: Context,
    private val level: Lesson,
    val packageSize: Int,
    val skiped: Boolean,
    val prize: Int,
    val finishLevel: FinishLevel
) : Dialog(
    context
), View.OnClickListener {
    private val lengthManager: LengthManager = LengthManager(context)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setCanceledOnTouchOutside(true)
        setContentView(R.layout.dialog_next_level)
        window!!.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
        val prizeTextView = findViewById<View>(R.id.prizeTextView) as TextView
        val scarf = findViewById<ImageView>(R.id.scarf_image_view)
        if (!level.resolved && !skiped) {
            val prizeString: String = if (prize == 30) "+۳۰" else if (prize == 10) "+۱۰" else "+۵"
            customizeTextView(prizeTextView, prizeString, lengthManager.getLevelAuthorTextSize())
            Glide.with(context).load(R.drawable.scarf).into(scarf)
        } else {
            prizeTextView.visibility = View.GONE
            scarf.visibility = View.GONE
        }
        val width = (lengthManager.screenWidth * 0.8f).toInt()
        val height = lengthManager.getHeightWithFixedWidth(
            R.drawable.dialog_background, (lengthManager.screenWidth * 0.8f).toInt()
        )
        val container: ConstraintLayout = findViewById(R.id.dialog_container)


//        container.setMaxWidth(width);
        container.minWidth = width
        container.minHeight = height
        val background = findViewById<ImageView>(R.id.dialog_background)
        val lp = background.layoutParams as ConstraintLayout.LayoutParams
        lp.width = width
        lp.height = height
        background.layoutParams = lp
        //        UiUtil.setWidth(background, width);
//        UiUtil.setHeight(background, height);
        Glide.with(getContext()).load(R.drawable.dialog_background).into(background)
        val nextButton = findViewById<ImageView>(R.id.next_level_button)
        Glide.with(getContext()).load(R.drawable.next_button_dialog).into(nextButton)
        nextButton.setOnClickListener(this)
        val homeButton = findViewById<ImageView>(R.id.home_button)
        Glide.with(getContext()).load(R.drawable.hoem_button).into(homeButton)
        homeButton.setOnClickListener(this)
        val tick = findViewById<ImageView>(R.id.tick)
        Glide.with(getContext()).load(R.drawable.tick).into(tick)
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.home_button -> finishLevel.Home()
            R.id.next_level_button -> finishLevel.NextLevel()
        }
        dismiss()
    }

    private fun customizeTextView(textView: TextView, label: String, textSize: Float) {
        textView.text = label
        textView.setTextSize(TypedValue.COMPLEX_UNIT_PX, lengthManager.getStoreItemFontSize())
        textView.setTextColor(Color.WHITE)
        textView.setShadowLayer(1f, 2f, 2f, Color.BLACK)
        //        textView.setTypeface(FontsHolder.getSansBold(textView.getContext()));
    }

}